### PR TITLE
add instance segmentation per_sample_transform

### PR DESCRIPTION
**Pull request by MrShevan**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/lightning-flash/issues/930

## Before submitting
- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes?
- [x] Did you write any **new necessary tests**? [not needed for typos/docs]
- [x] Did you verify **new and existing tests pass** locally with your changes?
- [x] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
- [x] Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by MrShevan** [comment]
@ethanwharris Hi, may I ask for help? )
I created all necessary tests for my new code changes. But It's strange that Codecov reports: `will decrease coverage by 19.69%.`

Could you approve a workflow for test running?

**Review by ethanwharris — APPROVED** [review]
@MrShevan This looks awesome! Don't worry about the tests, I think something is going wrong unrelated to your PR :smiley:

File: tests/image/instance_segmentation/test_data.py [hunk 1]
```diff
@@ -43,3 +46,31 @@ def test_image_detector_data_from_folders(tmpdir):
...
+            "scores": [0.5, 0.5],
+        },
+        DataKeys.METADATA: {"size": (224, 224)},
+    }
+
+    output_transform_cls = InstanceSegmentationOutputTransform()
+    data = output_transform_cls.per_sample_transform(sample)
+
+    assert data["masks"][0].size() == (224, 224)
+    assert data["masks"][1].size() == (224, 224)
```
Comment (krshrimali) [inline tests/image/instance_segmentation/test_data.py:76]: Nice test! Thank you for adding this as well.

**Review by krshrimali — APPROVED** [review]
Thanks, @MrShevan for the PR! LGTM.

**Comment by krshrimali** [comment]
@MrShevan - I guess we can merge this :) I'm going to merge this for now, once all tests are done. Thank you for your contribution.

File: tests/image/instance_segmentation/test_data.py (hunk 1 above)
Comment (MrShevan) [reply tests/image/instance_segmentation/test_data.py:76]: Thank you for your comment :)

**Comment by MrShevan** [comment]
@ethanwharris @krshrimali Thanks for your review ! It was a pleasure for me to work on this PR :)

**Pull request merged**